main/test/flix/Test.Exp.Match.Record.flix

```diff
@@ -0,0 +1,73 @@
+/*
+ *  Copyright 2023 Jakob Schneider Villumsen
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+mod Test.Exp.Match.Record {
+
+    @test
+    def testMatchRecord01(): Bool = match { } {
+        case _ => true
+    }
+
+    @test
+    def testMatchRecord02(): Bool = match { } {
+        case { } => true
+        case _   => false
+    }
+
+    @test
+    def testMatchRecord03(): Bool = match { x = 1 } {
+        case { x } => x == 1
+    }
+
+    @test
+    def testMatchRecord04(): Bool = match { x = 1 } {
+        case { x = 1 } => true
+    }
+
+    @test
+    def testMatchRecord05(): Bool = match { x = 1 } {
+        case { x = 2 } => false
+        case { x = 1 } => true
+        case _ => false
+    }
+
+    @test
+    def testMatchRecord06(): Bool = match { x = 3 } {
+        case { x = 1 } => false
+        case { x = 2 } => false
+        case { x = 3 } => true
+    }
+
+    @test
+    def testMatchRecord07(): Bool = match { x = 3 } {
+        case { x = 1 } => false
+        case { x = 2 } => false
+        case { x = 3 } => true
+        case { x = 4 } => false
+    }
+
+    @test
+    def testMatchRecord08(): Bool = match { x = 1 } {
+        case { x | r } => x == 1
+    }
+
+    @test
+    def testMatchRecord09(): Bool =
+        let rec = { c = "hello" };
+        let { c } = rec;
+        c == "hello"
+
```

Comment:
I think tests should focus on destructing records and then testing that their components are correct.

E.g. more like `testMatchRecord03`